components/automate-ui/src/app/entities/users/user.model.ts

```diff
@@ -1,7 +1,8 @@
 export interface User {
   id: string;
   name: string;
-  username: string;
+  membership_id: string;
```

Comment:
We need some further changes here so we can still handle `v1` users, otherwise the Users UI breaks when you reset to v1. 
Since v1 and v2 users use different unique identifiers (v1 uses a GUID, v2 uses string id), we have to convert all returned users into one standard user interface that includes both identifiers.

Here's how we did it with teams:
- two interfaces, one for v1, one for v2: https://github.com/chef/automate/blob/master/components/automate-ui/src/app/entities/teams/team.model.ts
- `versionize` function to convert v1 users to v2 interface
https://github.com/chef/automate/blob/master/components/automate-ui/src/app/entities/teams/team.effects.ts#L63
- versionize is called anywhere we get user objects back from the API
https://github.com/chef/automate/blob/master/components/automate-ui/src/app/entities/teams/team.requests.ts#L23

Reply:
Hmmm I was trying to avoid V1 details anywhere besides requests. That's a ton of functionality to support a super specific situation. What if we just do something similar to what we are going to do with the project filters: cache the current version, and if it changes between v1 and v2 or vise versa, clear the store and reload? Seems like a much cleaner solution from a code perspective for something we only ever expect to happen once or twice in the lifetime of a customer's install.

Reply:
Oh also I updated everywhere in the app to use v2 identifiers, even if it's on V1. That's only "breaking" in the sense that the URL changes, but that's a change that's going to have to happen eventually regardless and a change we've already made in a few spots if memory serves.

Reply:
I updated the whole app to use the V2 user model. It's a "breaking" change in the sense that all the user UI routes will now use ID instead of the GUID, but to get the new code, they need to reload the app, so all the links within the app with be correct.

There are actually issues with resetting for teams. If you are on a teams details page and you reset to v1, you'll get this error:

<img width="1436" alt="Screen Shot 2019-05-08 at 1 21 44 PM" src="https://user-images.githubusercontent.com/1899715/57405863-5237d500-7194-11e9-9467-5babb28070fd.png">

You also get that error if you are on the list view, reset, then click a team, because the IDs are different between v1 and v2.

This keeps all those details internal to `requests` and keeps the model consistent between v1 an v2. I'm not seeing any issues for users on this branch and the above exists on master.

Reply:
ah 💡so we can just switch the user ID since we're getting the same information from the users API whether we're on v1 or v2. but teams come from different tables depending on the version so we have different data